tls/s2n_config.c

```diff
@@ -886,7 +886,9 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config,
 
     if (intro_time_in_seconds_from_epoch == 0) {
         uint64_t now = 0;
-        POSIX_GUARD(config->wall_clock(config->sys_clock_ctx, &now));
+
+        POSIX_ENSURE(config->wall_clock(config->sys_clock_ctx, &now) >= S2N_SUCCESS, S2N_ERR_CANCELLED);
+
         session_ticket_key->intro_timestamp = now;
```

Comment:
Nit: I don't think these extra newlines you added help with readability. Can you remove them?